test_autoarray/geometry/test_geometry_util.py

```diff
@@ -3,1087 +3,1079 @@
+        scaled_coordinates_2d=(0.01, -0.01),
+        shape_native=(2, 2),
+        pixel_scales=(2.0, 2.0),
+    )
 
-        assert pixel_coordinates == (0, 1)
+    assert pixel_coordinates == (0, 0)
 
-        pixel_coordinates = aa.util.geometry.pixel_coordinates_2d_from(
-            scaled_coordinates_2d=(3.01, 2.99),
-            shape_native=(2, 2),
-            pixel_scales=(2.0, 2.0),
-            origins=(1.0, 1.0),
-        )
+    pixel_coordinates = aa.util.geometry.pixel_coordinates_2d_from(
+        scaled_coordinates_2d=(2.01, 0.01),
+        shape_native=(2, 2),
+        pixel_scales=(2.0, 2.0),
+    )
 
-        assert pixel_coordinates == (0, 1)
+    assert pixel_coordinates == (0, 1)
 
-        pixel_coordinates = aa.util.geometry.pixel_coordinates_2d_from(
-            scaled_coordinates_2d=(1.01, 1.01),
-            shape_native=(2, 2),
-            pixel_scales=(2.0, 2.0),
-            origins=(1.0, 1.0),
-        )
+    pixel_coordinates = aa.util.geometry.pixel_coordinates_2d_from(
+        scaled_coordinates_2d=(2.01, 1.99),
+        shape_native=(2, 2),
+        pixel_scales=(2.0, 2.0),
+    )
 
-        assert pixel_coordinates == (0, 1)
+    assert pixel_coordinates == (0, 1)
 
-        pixel_coordinates = aa.util.geometry.pixel_coordinates_2d_from(
-            scaled_coordinates_2d=(1.01, 2.99),
-            shape_native=(2, 2),
-            pixel_scales=(2.0, 2.0),
-            origins=(1.0, 1.0),
-        )
+    pixel_coordinates = aa.util.geometry.pixel_coordinates_2d_from(
+        scaled_coordinates_2d=(0.01, 0.01),
+        shape_native=(2, 2),
+        pixel_scales=(2.0, 2.0),
+    )
 
-        assert pixel_coordinates == (0, 1)
+    assert pixel_coordinates == (0, 1)
 
-        pixel_coordinates = aa.util.geometry.pixel_coordinates_2d_from(
-            scaled_coordinates_2d=(0.99, -0.99),
-            shape_native=(2, 2),
-            pixel_scales=(2.0, 2.0),
-            origins=(1.0, 1.0),
-        )
+    pixel_coordinates = aa.util.geometry.pixel_coordinates_2d_from(
+        scaled_coordinates_2d=(0.01, 1.99),
+        shape_native=(2, 2),
+        pixel_scales=(2.0, 2.0),
+    )
+
+    assert pixel_coordinates == (0, 1)
 
-        assert pixel_coordinates == (1, 0)
+    pixel_coordinates = aa.util.geometry.pixel_coordinates_2d_from(
+        scaled_coordinates_2d=(-0.01, -1.99),
+        shape_native=(2, 2),
+        pixel_scales=(2.0, 2.0),
+    )
+
+    assert pixel_coordinates == (1, 0)
 
-        pixel_coordinates = aa.util.geometry.pixel_coordinates_2d_from(
-            scaled_coordinates_2d=(0.99, 0.99),
-            shape_native=(2, 2),
-            pixel_scales=(2.0, 2.0),
-            origins=(1.0, 1.0),
-        )
+    pixel_coordinates = aa.util.geometry.pixel_coordinates_2d_from(
+        scaled_coordinates_2d=(-0.01, -0.01),
+        shape_native=(2, 2),
+        pixel_scales=(2.0, 2.0),
+    )
+
+    assert pixel_coordinates == (1, 0)

```

Comment:
I've never seen a better use case for [parametrize](https://docs.pytest.org/en/6.2.x/parametrize.html)!